Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs

```diff
@@ -283,32 +298,79 @@ private async void OnOnlineModeDropdownChanged(OnlineMode value)
             }
             else if (value == OnlineMode.UnityRelay)
             {
-                Exception caughtException = null;
-                Task<List<UnityRegion>> task = null;
-                try
+                m_ReconnectingImage.SetActive(true);
+                m_InputField.gameObject.SetActive(false);
+                m_PortInputField.gameObject.SetActive(false);
+                m_ConfirmationButton.gameObject.SetActive(false);
+                m_MainText.text = "Waiting for Unity Relay Health Check...";
+
+                if (m_UnityRelayHealthCheck == null || m_UnityRelayHealthCheck.IsCompleted)

```

Comment:
1) let's doc this for the flow
2) health check should be done every new game. service can go down between two games.